# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/extensions/webidl-api/ExtensionEventListener.h
# Commit: 7802e5f1bd13
# Full Hash: 7802e5f1bd1395f19f2d78636787c41f64192bd7
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2024-04-19 15:22:44
# Regressor Bug: 1769913
# File Overlap Count: 3
# Description:
#   Bug 1769913 - P3 Remove WorkerRunnable::mWorkerPrivate. r=dom-worker-reviewers,asuth
#   
#   WorkerRunnable no longer keeps a raw pointer(mWorkerPrivate) for the associated WorkerPrivate in this patch.
#   Removing the WorkerRunnable::mWorkerPrivate needs to fix the following problems.
#   
# ==============================================================================

diff -r ffc69aa8f2c3 -r 7802e5f1bd13 toolkit/components/extensions/webidl-api/ExtensionEventListener.h
--- a/toolkit/components/extensions/webidl-api/ExtensionEventListener.h	Fri Apr 19 09:41:57 2024 +0000
+++ b/toolkit/components/extensions/webidl-api/ExtensionEventListener.h	Fri Apr 19 09:41:58 2024 +0000
@@ -134,8 +134,7 @@
       UniquePtr<dom::StructuredCloneHolder> aArgsHolder,
       ListenerCallOptions* aCallOptions,
       RefPtr<dom::Promise> aPromiseRetval = nullptr)
-      : WorkerThreadRunnable(aExtensionEventListener->GetWorkerPrivate(),
-                             "ExtensionListenerCallWorkerRunnable"),
+      : WorkerThreadRunnable("ExtensionListenerCallWorkerRunnable"),
         mListener(aExtensionEventListener),
         mArgsHolder(std::move(aArgsHolder)),
         mPromiseResult(std::move(aPromiseRetval)),
